### PR TITLE
[tests-only][full-ci]add e2e test for breadcrumb navigation

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
In this PR,
- removed the webUI test scenario related to breadcrumb navigation
- added E2E test to navigate throught breadcrumb
- different variety of folder names has been used as suggested from comment: https://github.com/owncloud/web/issues/10259#issuecomment-2011540807

## Related Issue

- https://github.com/owncloud/web/issues/10259

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 1]
```diff
@@ -139,6 +146,23 @@ export const clickResource = async ({
...
+  const folder = buildXpathLiteral(resource)
+  const itemId = await page
+    .locator(util.format(breadcrumbResourceSelector, folder))
+    .getAttribute('data-item-id')
+  await Promise.all([
+    page.waitForResponse(
+      (resp) =>
+        resp.url().endsWith(encodeURIComponent(resource)) ||
+        resp.url().endsWith(itemId) ||
+        resp.url().endsWith(encodeURIComponent(itemId))
```
Comment (ScharfViktor) [inline tests/e2e/support/objects/app-files/resource/actions.ts:159]: please add which method do you expect here and which status code

File: tests/e2e/support/utils/locator.ts [hunk 2]
```diff
@@ -13,3 +13,16 @@ export const waitForEvent = (locator: Locator, type: keyof SVGElementEventMap):
...
+    // if we don't have any single quotes, then wrap them with single quotes
+    return "'" + value + "'"
+  } else if (!value.includes('"')) {
+    // if we don't have any double quotes, then wrap them with double quotes
+    return '"' + value + '"'
+  } else {
+    // use concat to find the literal in the xpath if they contain both quotes
+    return "concat('" + value.replace(/'/g, "',\"'\",'") + "')"
+  }
+}
```
Comment (ScharfViktor) [inline tests/e2e/support/utils/locator.ts:28]: should we use `switch` here?

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 3]
```diff
@@ -25,8 +26,11 @@ export const fileRow =
   '//ancestor::*[(contains(@class, "oc-tile-card") or contains(@class, "oc-tbody-tr"))]'
 export const resourceNameSelector =
   ':is(#files-space-table, .oc-tiles-item, #files-shared-with-me-accepted-section, .files-table) [data-test-resource-name="%s"]'
+// following breadcrumb selectors is passed to buildXpathLiteral function as the content to be insetred might contain quotes
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/resource/actions.ts:29]:
```suggestion
// following breadcrumb selectors is passed to buildXpathLiteral function as the content to be inserted might contain quotes
```

**Review by saw-jan — APPROVED** [review]
LGTM :+1:

**Comment by S-Panta** [comment]
LGTM

**Review by SagarGi — APPROVED** [review]
LGTM :+1:

**Pull request merged**